### PR TITLE
[INJI-207] Sharing issues

### PR DESCRIPTION
**Pull request by Alka1703**

### CODE REVIEW

File: android/src/main/java/io/mosip/tuvali/ble/central/impl/Controller.kt [hunk 1]
```diff
@@ -159,10 +159,12 @@ class Controller(val context: Context) {
     messageSender.sendMessage(failedMessage)
   }
 
-  private fun onWriteSuccess(device: BluetoothDevice, charUUID: UUID) {
-    val writeSuccessMessage = WriteSuccessMessage(device, charUUID)
 
-    messageSender.sendMessage(writeSuccessMessage)
+  private fun onWriteSuccess(device: BluetoothDevice?, charUUID: UUID) {
```
Comment (vharsh) [inline android/src/main/java/io/mosip/tuvali/ble/central/impl/Controller.kt:163]: Why did the signature have to change?

Is there a case where the peripheralDevice is not null(at a time t1) and when we get the callback(at a time `t1+X`) the Device is already disconnected and an exception can be thrown if the `peripheralDevice` reference is used at a time `t1+X+Y` ?

File: android/src/main/java/io/mosip/tuvali/ble/central/impl/GattClient.kt [hunk 2]
```diff
@@ -179,7 +177,7 @@ class GattClient(var context: Context) {
     serviceUuid: UUID,
     charUUID: UUID,
     data: ByteArray,
-    onSuccess: (BluetoothDevice, UUID) -> Unit,
+    onSuccess: (BluetoothDevice?, UUID) -> Unit,
     onFailed: (BluetoothDevice, UUID, Int) -> Unit
```
Comment (vharsh) [inline android/src/main/java/io/mosip/tuvali/ble/central/impl/GattClient.kt]: For consistency sake, should onFailed fun not take an optional of `BluetoothDevice` object ?

File: android/src/main/java/io/mosip/tuvali/ble/central/impl/GattClient.kt [hunk 3]
```diff
@@ -191,14 +189,13 @@ class GattClient(var context: Context) {
...
     writeChar?.value = data
     writeChar?.writeType = BluetoothGattCharacteristic.WRITE_TYPE_NO_RESPONSE
+    this.onWriteSuccess = onSuccess
+    this.onWriteFailed = onFailed
     val status = bluetoothGatt?.writeCharacteristic(writeChar)
 
     if (status == false) {
       return onFailed(device, charUUID, GATT_FAILURE)
     }
-
```
Comment (vharsh) [inline android/src/main/java/io/mosip/tuvali/ble/central/impl/GattClient.kt:199]: 👍🏼

**Review by vharsh — COMMENTED** [review]
sharing some doubts

File: android/src/main/java/io/mosip/tuvali/ble/central/impl/GattClient.kt (hunk 2 above)
Comment (tilak-puli) [reply android/src/main/java/io/mosip/tuvali/ble/central/impl/GattClient.kt]: removed bluetooth device from both onSuccess and onFailed

Comment (PuBHARGAVI) [reply android/src/main/java/io/mosip/tuvali/ble/central/impl/GattClient.kt]: when calling onWriteSuccess method before we used to check if peripheral device and characteristics are available or not then we are calling the method but now we are checking if characteristic is available or not then we are calling onWriteSuccess method and inside that we are checking if peripheral is available or not that is why we are adding "?" in onWriteSuccess method but this flow is not changed for onWriteFailed

File: android/src/main/java/io/mosip/tuvali/ble/central/impl/Controller.kt (hunk 1 above)
Comment (tilak-puli) [reply android/src/main/java/io/mosip/tuvali/ble/central/impl/Controller.kt:163]: We haven't seen any disconnect or connect logs whenever this happens. 

we are still researching on why this happening but for now we have removed null check on device as we still want to send success.

**Pull request merged**